### PR TITLE
nixos,doc: use networkmanager and nmtui on the non-graphical installer for better UX

### PR DESCRIPTION
**Pull request by blaggacao**
- nixos,doc: use ifwifi to improve installation device UX

### CODE REVIEW

**Comment by blaggacao** [comment]
doesn't even look too bad...

on a random project of mine, where I enabled `ifwifi`

## before

```console
❯ dust ./result/iso
1021M   ┌── bootstrap-hive-from-queen-22.05pre-git-x86_64-linux.isonixos.iso│███████████████ │ 100%
1021M ┌─┴ iso  
```

## after

```console
❯ dust ./result/iso
1022M   ┌── bootstrap-hive-from-queen-22.05pre-git-x86_64-linux.isonixos.iso│███████████████ │ 100%
1022M ┌─┴ iso     
```

File: nixos/doc/manual/from_md/installation/installing.chapter.xml [hunk 1]
```diff
@@ -255,54 +255,35 @@ $ sudo -i
...
-        first start wpa_supplicant with
-        <literal>sudo systemctl start wpa_supplicant</literal>, then run
-        <literal>wpa_cli</literal>. For most home networks, you need to
-        type in the following commands:
+        On the minimal installer, the configuration must be perfomed on
+        the command line. To configure the wifi, first start the
+        NetworkManager with
+        <literal>sudo systemctl start NetworkManager</literal>, then run
+        <literal>sudo ifwifi scan</literal> to scan for available
+        networks. To connext to one of the networks, you need to type in
```
Comment (bjornfor) [inline nixos/doc/manual/from_md/installation/installing.chapter.xml:263]: Typo: connext

**Comment by bjornfor** [comment]
Did you consider nmtui?

**Comment by blaggacao** [comment]
@bjornfor yes, but it is a bit stressful, as well. The navigation isn't quite as intuitive for the simple case. I should probably mention it in the docs for the edu/enterprise case.

What I really want is to implement dynamic upstream CLI completion so that you can just do `ifwifi<TAB>` and do the right thing.

This "requirement" is based on my observation that the moment you're on the terminal without internet and no `nix` command is working out of the box (and you can't get additional.packages from the cache) was always generally quite stressful for me. ("Will it work?" "Can I remember how to connect?"). Relevant here: it doesn't happen that often.

So I want to make the happy path as straight forward as possible.

Several upstream PRs to improve `iwifi` are already created and should percolate into `nixpkgs` over time.

File: nixos/doc/manual/installation/installing.chapter.md [hunk 2]
```diff
@@ -161,49 +161,13 @@ so even in a non-graphical session. If you prefer to configure the
...
-<3>CTRL-EVENT-CONNECTED - Connection to 32:85:ab:ef:24:5c completed [id=0 id_str=]
-```
+On the minimal installer, the configuration must be perfomed on the
+command line. To configure the wifi, first start the NetworkManager
+with `sudo systemctl start NetworkManager`, then run
+`sudo nmtui connect` to guide you through the process.
 
-you can now leave `wpa_cli` by typing `quit`.
+For enterprise networks, for example *eduroam*, instead use the more
+complete `sudo nmtui` interface to configure the Wifi.
```
Comment (bjornfor) [inline nixos/doc/manual/installation/installing.chapter.md:170]: Wifi -> wifi? Or go all-in with `Wi-Fi`?

Comment (blaggacao) [reply nixos/doc/manual/installation/installing.chapter.md:170]: Fixed & force-pushed

**Comment by blaggacao** [comment]
### On this branch:
```
nixpkgs on  mod-doc-ifwifi-installer [$]
❯ dust ./result/iso
4.1G   ┌── nixos-minimal-22.11pre130979.gfedcba-aarch64-linux.iso

# in bytes
❯ l -B /nix/store/gxc939amrx95d3gahdfwv04k4r9qgkaq-nixos-minimal-22.11pre130979.gfedcba-x86_64-linux.iso/iso/nixos-minimal-22.11pre130979.gfedcba-x86_64-linux.iso
Permissions          Size User      Date Modified Name
.r--r--r--  4,408,750,080 blaggacao 31 Dec  1969  /nix/store/gxc939amrx95d3gahdfwv04k4r9qgkaq-nixos-minimal-22.11pre130979.gfedcba-x86_64-linux.iso/iso/nixos-minimal-22.11pre130979.gfedcba-x86_64-linux.iso
```

### The Commit Reverted:
```
nixpkgs on  mod-doc-ifwifi-installer [$⇡] took 4s
❯ g log HEAD
commit 7ba81600d1ffdf27ba9d22dc0d43b716a7eafe28 (HEAD -> mod-doc-ifwifi-installer)
Author: David Arnold <david.arnold@iohk.io>
Date:   Mon Nov 7 22:09:17 2022 -0500

    Revert "nixos,doc: use networkmanager and nmtui to improve installation device UX"

    This reverts commit 379fb7b346b2113d1efdcf6fe3e68d4ab34a704b.
```

->
```
❯ dust ./result/iso
3.9G   ┌── nixos-minimal-22.11pre130979.gfedcba-aarch64-linux.iso

❯ l -B ./result/iso
Permissions          Size User      Date Modified Name
.r--r--r--  4,201,213,952 blaggacao 31 Dec  1969  nixos-minimal-22.11pre130979.gfedcba-aarch64-linux.iso
```

**Comment by blaggacao** [comment]
After more consideration, i'm going to close this, since 200Mb seems too much. Rather `ifwifi` should be improved to work with `wpa_supplicant` or `iwctl`

**Pull request closed without merging**